gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -222,27 +225,33 @@ public static ClientContext create(StubSettings settings) throws IOException {
    * Project Id.
    */
   private static Map<String, String> getHeadersFromSettings(StubSettings settings) {
-    ImmutableMap.Builder<String, String> headersBuilder = ImmutableMap.builder();
-    if (settings.getQuotaProjectId() != null) {
-      headersBuilder.put(QUOTA_PROJECT_ID_HEADER_KEY, settings.getQuotaProjectId());
-      for (Map.Entry<String, String> entry : settings.getHeaderProvider().getHeaders().entrySet()) {
-        if (entry.getKey().equals(QUOTA_PROJECT_ID_HEADER_KEY)) {
-          continue;
-        }
-        headersBuilder.put(entry);
+    // Resolve conflicts when merging headers from multiple sources
+    Map<String, String> userHeaders = settings.getHeaderProvider().getHeaders();
+    Map<String, String> internalHeaders = settings.getInternalHeaderProvider().getHeaders();
+    Map<String, String> conflictResolution = new HashMap<>();
+
+    Set<String> conflicts = Sets.intersection(userHeaders.keySet(), internalHeaders.keySet());
+    for (String key : conflicts) {
+      if ("user-agent".equals(key)) {
+        conflictResolution.put(key, userHeaders.get(key) + " " + internalHeaders.get(key));
```

Comment:
I'm not sure you can get away with this. This doesn't seem necessarily consistent with the HTTP spec: https://tools.ietf.org/html/rfc7231#section-5.5.3

Reply:
Which part is inconsistent? Using the rfc example of `User-Agent: CERN-LineMode/2.15 libwww/2.17b3`

`libwww/2.17b3` is the internal header and `CERN-LineMode/2.15` is the external